showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -290,16 +290,18 @@ class ShowkaseProcessor @JvmOverloads constructor(
         }
     }
         .distinctBy {
-            // We also ensure that the component groupName and the component name are unique so 
-            // that they don't show up twice in the browser app.
+            // We also ensure that the component groupName and the component name are unique so
+            // that they don't show up twice in the browser app. This also de-duplicates based
+            // on the fully qualified function name to support categorization with additional
+            // fields (e.g. tags, extraMetadata, etc) on custom browsers.
             if (it.componentIndex != null) {
-                "${it.showkaseName}_${it.showkaseGroup}_${it.showkaseStyleName}_${it.componentIndex}"
+                "${it.fqPrefix}_${it.showkaseName}_${it.showkaseGroup}_${it.showkaseStyleName}_${it.componentIndex}"
```

Comment:
we already do the `${it.packageName}_${it.enclosingClassName}_${it.elementName}` dedpuing above. Why do we need to do it again here 🤔

Reply:
@vinaygaba If I don't include it here as well it will remove ones that only differ by `tags` or `extraMetadata`. I tried to change this to do them in one `distinctBy` step but that messed up where a `@Preview` annotation has the same group/style as the `@ShowkaseComposable` annotation so I left it as two steps.

Reply:
I see!